src/firebase_functions/options.py

```diff
@@ -0,0 +1,316 @@
+    These are included in the Access-Control-Allow-Methods response headers
+    to the preflight OPTIONS requests.
+    """
+
+
+class Memory(int, _enum.Enum):
+    """
+    Available memory options supported by Cloud Functions.
+    """
+
+    MB_128 = 128
+    MB_256 = 256
+    MB_512 = 512
+    GB_1 = 1 << 10
+    GB_2 = 2 << 10
+    GB_4 = 4 << 10
+    GB_8 = 8 << 10
+    GB_16 = 16 << 10
+    GB_32 = 32 << 10
+
+
+class SupportedRegion(str, _enum.Enum):
+    """
+    All regions supported by Cloud Functions v2.
+    """
+
+    ASIA_NORTHEAST1 = "asia-northeast1"
+    EUROPE_NORTH1 = "europe-north1"
+    EUROPE_WEST1 = "europe-west1"
+    EUROPE_WEST4 = "europe-west4"
+    US_CENTRAL1 = "us-central1"
+    US_EAST1 = "us-east1"
+    US_WEST1 = "us-west1"
+
+
+@_dataclasses.dataclass(frozen=True)
+class _CommonOptions:
+    """
+    _CommonOptions are options that can be set on any function or globally.
+    Internal use only.
+    """
+
+    region: SupportedRegion | str | _Sentinel | None = None
+    """
+    Region where functions should be deployed.
+    HTTP functions can override and specify more than one region.
+    """
+
+    memory: int | Memory | Expression[int] | _Sentinel | None = None
+    """
+    Amount of memory to allocate to a function.
+    A value of USE_DEFAULT restores the defaults of 256MB.
+    """
+
+    timeout_sec: int | Expression[int] | _Sentinel | None = None
+    """
+    Timeout for the function in sections, possible values are 0 to 540.
+    HTTPS functions can specify a higher timeout.
+    A value of USE_DEFAULT restores the default of 60s
+    The minimum timeout for a gen 2 function is 1s. The maximum timeout for a
+    function depends on the type of function: Event handling functions have a
+    maximum timeout of 540s (9 minutes). HTTPS and callable functions have a
+    maximum timeout of 36,00s (1 hour). Task queue functions have a maximum
+    timeout of 1,800s (30 minutes)
+    """
+
+    min_instances: int | Expression[int] | _Sentinel | None = None
+    """
+    Min number of actual instances to be running at a given time.
+    Instances will be billed for memory allocation and 10% of CPU allocation
+    while idle.
+    A value of USE_DEFAULT restores the default min instances.
+    """
+
+    max_instances: int | Expression[int] | _Sentinel | None = None
+    """
+    Max number of instances to be running in parallel.
+    A value of USE_DEFAULT restores the default max instances.
+    """
+
+    concurrency: int | Expression[int] | _Sentinel | None = None
+    """
+    Number of requests a function can serve at once.
+    Can only be applied to functions running on Cloud Functions v2.
+    A value of USE_DEFAULT restores the default concurrency (80 when CPU >= 1, 1 otherwise).
+    Concurrency cannot be set to any value other than 1 if `cpu` is less than 1.
+    The maximum value for concurrency is 1,000.
+    """
+
+    cpu: int | _typing.Literal["gcf_gen1"] | _Sentinel = "gcf_gen1"
```

Comment:
This should default to None. (which the CLI will default to the maximum of "gcf_gen1" and 1.